cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -253,17 +300,23 @@ Result<Datum> GroupByTest(const std::vector<Datum>& arguments,
     idx = idx + 1;
   }
   if (use_exec_plan) {
-    return GroupByUsingExecPlan(arguments, keys, internal_aggregates, use_threads,
-                                small_chunksize_context(use_threads));
+    return GroupByUsingExecPlan(arguments, keys, segment_keys, internal_aggregates,
+                                use_threads, small_chunksize_context(use_threads));
   } else {
-    return internal::GroupBy(arguments, keys, internal_aggregates, use_threads,
-                             default_exec_context());
+    return AlternatorGroupBy(arguments, keys, segment_keys, internal_aggregates,
+                             use_threads, default_exec_context());
   }
 }
 
-}  // namespace
+Result<Datum> GroupByTest(const std::vector<Datum>& arguments,
+                          const std::vector<Datum>& keys,
+                          const std::vector<TestAggregate>& aggregates, bool use_threads,
+                          bool use_exec_plan = false) {
+  return GroupByTest(arguments, keys, {}, aggregates, use_threads, use_exec_plan);
+}
 
-TEST(Grouper, SupportedKeys) {
+template <typename GroupClass>
+void test_group_class_supported_keys() {
```

Comment:
> Why does this refactoring only replace some instances of Grouper with GroupClass?

This may be unintended. I'll look into this when I get back to this PR.